src/useStream.ts

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useRef, useState } from 'react';
+import { ConduitClient, type ConduitClientConfig } from './client';
+
+export function useStream<T>(config: ConduitClientConfig<T>, enabled: boolean) {
+  const clientRef = useRef<ConduitClient<T> | null>(null);
+  const isConnectingRef = useRef(false);
+  const [isConnected, setIsConnected] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const [messages, setMessages] = useState<T[]>([]);
+
```

Comment:
One thing you'll need to be careful about here is if the `baseConduitUrl` and `startStreamUrl` change in the hook.

Ideally the hook should reactively disconnect and reconnect with the new url.

In that case you probably do want the client to be a state `const [client, setClient] = ...`. This way the event handlers can be re-assigned if the client changes.

```tsx
useEffect(() => {
  client.on('event', onEvent);
  return () => client.off('event', onEvent);
), [client, onEvent]); 
```

So this useEffect will fire when the client changes